### PR TITLE
Add tests for PaymentManager

### PR DESCRIPTION
**Pull request by malinajirka**
This PR adds missing unit tests for classes in `com.woocommerce.android.cardreader.internal.payments` package. These classes were introduced in https://github.com/woocommerce/woocommerce-android/pull/3794.

This PR exceeds the recommended limit on number of changes within a single PR. However, since 98% of the changes are just newly added unit tests I believe it's fine. Having said that, please let me know if you'd prefer me to split it into multiple PRs.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: libs/cardreader/src/test/java/CoroutinesUtils.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Delay
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlin.coroutines.resume
+
+fun <T> test(context: CoroutineContext = EmptyCoroutineContext, block: suspend CoroutineScope.() -> T) {
+    runBlocking(context, block)
```
Comment (hichamboushaba) [inline libs/cardreader/src/test/java/CoroutinesUtils.kt]: I think using `runBlockingTest` would be better, and more consistent with the rest of the tests in the app, in addition to the more [advantages](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/run-blocking-test.html) it brings, WDYT?

**Review by hichamboushaba — CHANGES_REQUESTED** [review]
Nice job working on this, and covering a lot of use cases.
I left one comment regarding the usage of `runBlocking`.

File: libs/cardreader/src/test/java/CoroutinesUtils.kt (hunk 1 above)
Comment (malinajirka) [reply libs/cardreader/src/test/java/CoroutinesUtils.kt]: TIL about runBlockingTest ;)
Fixed in https://github.com/woocommerce/woocommerce-android/pull/3796/commits/593eee875e841d94e563a9db7caf7fe5df3b7535 (deprecates `test`)
and 
https://github.com/woocommerce/woocommerce-android/pull/3796/commits/72c35da589acd24c2e4b782fea2e4ded8427c491 (replaces `test` with `runBlockingTest` in CardReaderModule)

**Comment by malinajirka** [comment]
Thanks @hichamboushaba for the review! ;) It's ready for another round.

**Review by hichamboushaba — APPROVED** [review]
Thanks for making the changes, LGTM 👍 
There are just some failing lint checks, feel free to merge once they are fixed.

**Pull request merged**